### ISSUE TITLE
Scoring results based on where keyword(s) appears in the search result entries.

### ISSUE THREAD

**Issue by janakaonline** [RedisLabsModules/RediSearch#565]
Hi, I'm having bit of a problem getting the results ordered correctly,

Consider following 4 entries like
Colombo Fort
Cafe Social Colombo
Colombo Fort Cafe
Colombo's Largest New Year's Eve Party at Waters Edge with Marians & Misty

When I do a FT.SEARCH <index> colombo*
Result come as,
1.Colombo's Largest New Year's Eve Party at Waters Edge with Marians & Misty
2.Colombo Fort
3.Cafe Social Colombo
4.Colombo Fort Cafe

No. 1 result has the score of 9 and rest is has score of 4. But shouldn't it give score based on position of the keyword appears on the result?

Basically, is there a way to get result as follows?
Colombo Fort
Colombo Fort Cafe
Colombo's Largest New Year's Eve Party at Waters Edge with Marians & Misty
Cafe Social Colombo

Highly appreciate if some one can nudge me in the right direction to fix this issue. :)

Thanks

**Comment by mnunberg** [comment]
I don't believe any of our existing algorithms can do this, but it is worth looking into.

Because of the way RediSearch stores index entries, this is quite non-trivial to do, because terms are 'flattened' in the index; that is, we do not know where in a given field the term is stored; only that it exists within a given field.

There is a way to implement this, if we use the term offsets (what we use for SLOP), and the offset map (used for highlighting).

**Comment by janakaonline** [comment]
Thanks @mnunberg , guess I have to implement it manually. Thanks for the clarification. :)
